File: redome/src/main/java/br/org/cancer/modred/controller/dto/LogisticaDTO.java
package br.org.cancer.modred.controller.dto;

import java.io.Serializable;
import java.time.LocalDate;
import java.util.List;

import br.org.cancer.modred.model.ArquivoVoucherLogistica;
import br.org.cancer.modred.model.CentroTransplante;
import br.org.cancer.modred.model.ContatoTelefonicoDoador;
import br.org.cancer.modred.model.EmailContatoDoador;
import br.org.cancer.modred.model.EnderecoContatoDoador;
import br.org.cancer.modred.model.TransporteTerrestre;

/**
 * Informações sobre o detalhe da logística necessária ao doador ou material.
 * 
 * @author Pizão.
 *
 */
public class LogisticaDTO implements Serializable {
	private static final long serialVersionUID = -3652795976912962282L;

	private Long idDoador;
	private Long pedidoLogisticaId;
	private Long tarefaId;
	
	private LocalDate dataInicio;
	private LocalDate dataFinal;
	
	private CentroTransplante centroColeta;
	
	private List<EnderecoContatoDoador> enderecos;
	private List<ContatoTelefonicoDoador> telefones;
	private List<EmailContatoDoador> emails;
	
	private List<TransporteTerrestre> transporteTerrestre;
	private List<ArquivoVoucherLogistica> aereos;
	private List<ArquivoVoucherLogistica> hospedagens;
	
	private Long rmr;
	
	private String nomeDoador;
	
	private String observacao;
	
	private Long dmr;

	
	public Long getIdDoador() {
		return idDoador;
	}

	public void setIdDoador(Long idDoador) {
		this.idDoador = idDoador;
	}

	public LocalDate getDataInicio() {
		return dataInicio;
	}

	public void setDataInicio(LocalDate dataInicio) {
		this.dataInicio = dataInicio;
	}

	public LocalDate getDataFinal() {
		return dataFinal;
	}

	public void setDataFinal(LocalDate dataFinal) {
		this.dataFinal = dataFinal;
	}

	public List<EnderecoContatoDoador> getEnderecos() {
		return enderecos;
	}

	public void setEnderecos(List<EnderecoContatoDoador> enderecos) {
		this.enderecos = enderecos;
	}

	public List<ContatoTelefonicoDoador> getTelefones() {
		return telefones;
	}

	public void setTelefones(List<ContatoTelefonicoDoador> telefones) {
		this.telefones = telefones;
	}

	public List<EmailContatoDoador> getEmails() {
		return emails;
	}

	public void setEmails(List<EmailContatoDoador> emails) {
		this.emails = emails;
	}

	public List<TransporteTerrestre> getTransporteTerrestre() {
		return transporteTerrestre;
	}

	public void setTransporteTerrestre(List<TransporteTerrestre> transporteTerrestre) {
		this.transporteTerrestre = transporteTerrestre;
	}

	public String getObservacao() {
		return observacao;
	}

	public void setObservacao(String observacao) {
		this.observacao = observacao;
	}

	public Long getPedidoLogisticaId() {
		return pedidoLogisticaId;
	}

	public void setPedidoLogisticaId(Long pedidoLogisticaId) {
		this.pedidoLogisticaId = pedidoLogisticaId;
	}

	public List<ArquivoVoucherLogistica> getAereos() {
		return aereos;
	}

	public void setAereos(List<ArquivoVoucherLogistica> aereos) {
		this.aereos = aereos;
	}

	public List<ArquivoVoucherLogistica> getHospedagens() {
		return hospedagens;
	}

	public void setHospedagens(List<ArquivoVoucherLogistica> hospedagens) {
		this.hospedagens = hospedagens;
	}

	public CentroTransplante getCentroColeta() {
		return centroColeta;
	}

	public void setCentroColeta(CentroTransplante centroColeta) {
		this.centroColeta = centroColeta;
	}

	public Long getTarefaId() {
		return tarefaId;
	}

	public void setTarefaId(Long tarefaId) {
		this.tarefaId = tarefaId;
	}

	public Long getRmr() {
		return rmr;
	}

	public void setRmr(Long rmr) {
		this.rmr = rmr;
	}

	public String getNomeDoador() {
		return nomeDoador;
	}

	public void setNomeDoador(String nomeDoador) {
		this.nomeDoador = nomeDoador;
	}

	public Long getDmr() {
		return dmr;
	}

	public void setDmr(Long dmr) {
		this.dmr = dmr;
	}
	
}

File: redome-front/src/app/shared/dto/logistica.dto.ts
import { BaseEntidade } from '../base.entidade';
import { EnderecoContatoDoador } from "../../doador/endereco.contato.doador";
import { ContatoTelefonicoDoador } from "../../doador/contato.telefonico.doador";
import { EmailContatoDoador } from "../../doador/email.contato.doador";
import { ArquivoVoucherLogistica } from "../model/arquivo.voucher.logistica";
import { CentroTransplante } from "../dominio/centro.transplante";
import { TransporteTerrestre } from '../model/transporte.terrestre';

/**
 * Informação dos detalhes da logística necessária a determinado pedido de logística.
 */
export class LogisticaDTO extends BaseEntidade {
	public jsonToEntity(res: any) {
		throw new Error("Method not implemented.");
	}
    private _idDoador: number;
	private _pedidoLogisticaId: number;
	private _tarefaId: number;

	private _dataInicio: Date;
	private _dataFinal: Date;

	private _enderecos: EnderecoContatoDoador[];
	private _telefones: ContatoTelefonicoDoador[];
	private _emails: EmailContatoDoador[];

	private _transporteTerrestre: TransporteTerrestre[];
	private _aereos: ArquivoVoucherLogistica[];
	private _hospedagens: ArquivoVoucherLogistica[];

	private _centroColeta: CentroTransplante;

	private _observacao: string;

	private _rmr:number;

	private _nomeDoador:string;

	private _dmr:number;


	public get idDoador(): number {
		return this._idDoador;
	}

	public set idDoador(value: number) {
		this._idDoador = value;
	}

	public get pedidoLogisticaId(): number {
		return this._pedidoLogisticaId;
	}

	public set pedidoLogisticaId(value: number) {
		this._pedidoLogisticaId = value;
	}

	public get dataInicio(): Date {
		return this._dataInicio;
	}

	public set dataInicio(value: Date) {
		this._dataInicio = value;
	}

	public get dataFinal(): Date {
		return this._dataFinal;
	}

	public set dataFinal(value: Date) {
		this._dataFinal = value;
	}

	public get enderecos(): EnderecoContatoDoador[] {
		return this._enderecos;
	}

	public set enderecos(value: EnderecoContatoDoador[]) {
		this._enderecos = value;
	}

	public get telefones(): ContatoTelefonicoDoador[] {
		return this._telefones;
	}

	public set telefones(value: ContatoTelefonicoDoador[]) {
		this._telefones = value;
	}

	public get emails(): EmailContatoDoador[] {
		return this._emails;
	}

	public set emails(value: EmailContatoDoador[]) {
		this._emails = value;
	}

	public get transporteTerrestre(): TransporteTerrestre[] {
		return this._transporteTerrestre;
	}

	public set transporteTerrestre(value: TransporteTerrestre[]) {
		this._transporteTerrestre = value;
	}

	public get aereos(): ArquivoVoucherLogistica[] {
		return this._aereos;
	}

	public set aereos(value: ArquivoVoucherLogistica[]) {
		this._aereos = value;
	}

	public get hospedagens(): ArquivoVoucherLogistica[] {
		return this._hospedagens;
	}

	public set hospedagens(value: ArquivoVoucherLogistica[]) {
		this._hospedagens = value;
	}

	public get centroColeta(): CentroTransplante {
		return this._centroColeta;
	}

	public set centroColeta(value: CentroTransplante) {
		this._centroColeta = value;
	}

	public get observacao(): string {
		return this._observacao;
	}

	public set observacao(value: string) {
		this._observacao = value;
	}

	public get tarefaId(): number {
		return this._tarefaId;
	}

	public set tarefaId(value: number) {
		this._tarefaId = value;
	}

	public get rmr(): number {
		return this._rmr;
	}

	public set rmr(value: number) {
		this._rmr = value;
	}

	public get nomeDoador(): string {
		return this._nomeDoador;
	}

	public set nomeDoador(value: string) {
		this._nomeDoador = value;
	}

	public get dmr(): number {
		return this._dmr;
	}

	public set dmr(value: number) {
		this._dmr = value;
	}


}
